File: server/matchmaking/src/main/java/com/boxtrotstudio/ghost/matchmaking/network/packets/SetNamePacket.java
package com.boxtrotstudio.ghost.matchmaking.network.packets;

import com.boxtrotstudio.ghost.matchmaking.Main;
import com.boxtrotstudio.ghost.matchmaking.network.PlayerClient;
import com.boxtrotstudio.ghost.matchmaking.network.TcpServer;
import com.boxtrotstudio.ghost.network.packet.Packet;
import com.boxtrotstudio.ghost.network.validate.DummyValidator;

import java.io.IOException;

public class SetNamePacket extends Packet<TcpServer, PlayerClient> {

    @Override
    public void onHandlePacket(PlayerClient client)  throws IOException {
        String name = consume(255).asString();

        if (Main.SESSION_VALIDATOR instanceof DummyValidator) {
            client.getPlayer().setName(name.trim());
            client.getPlayer().setPlayerID(name.trim().hashCode());
        }
    }
}

File: client/core/src/com/boxtrotstudio/ghost/client/network/packets/SetNamePacket.kt
package com.boxtrotstudio.ghost.client.network.packets

import com.boxtrotstudio.ghost.client.network.Packet
import com.boxtrotstudio.ghost.client.network.PlayerClient

class SetNamePacket : Packet<PlayerClient>() {

    override fun write(vararg args : Any) {
        var username = args[0] as String

        for (i in username.length..254) {
            username += " "
        }

        write(0x41.toByte())
        write(username)
        endTCP()
    }
}
